### ISSUE TITLE
Suggestions for genericity

### ISSUE THREAD

**Issue by tryingsomestuff** [AndyGrant/OpenBench#62]
Hi, 

I recently switch from OpenBench.py to Client.py, and this is now way easier for Minic use it thanks to all your recent additions.

Maybe you can have a look at this idea (https://github.com/tryingsomestuff/OpenBench/commit/52ddc72c7ae301617815ddb24ddd984eb90f73ed).

Two main stuff in here :
- different buildpath (where is the makefile) and outpath (where is the executable)
- use of exename inside config.py instead of passing it through makefile argument

Regards,

**Comment by AndyGrant** [comment]
**Makefile Path:**  I believe I have already done what you suggest, found here https://github.com/AndyGrant/OpenBench/commit/8adb23d974624ab513341cddd8d05e9511854851

**Exename:** This seems to not matter? In your diff you pass an exename to save as the output name, but then that information is immediately lost with ``os.rename(output, destination)``. It would seem that the only reason would be that a makefile lacks an ``EXE=`` hook, which I suppose is possible, but that is one thing that has been in (all? most?) of the engines I have looked at.

Just had a look at Minic and I see the ``.sh`` setup you have going on. Which explains your aversion to ``EXE=``.

**Outpath:** I suppose you have a point here. I can see a reason to dump binaries into a ``/dist/`` folder as you do. Particularly if you are using an IDE of some sort. Then again, that sounds like the sort of thing a makefile should have an argument to control that. 


I think ultimately, I don't want to make any of these changes, aside from the makefile path which I already did (Suggested by either you, or Rubichess author). OpenBench is fairly adaptable, but "technically" speaking your setup is far more complex to support than the engines on the main instance. Namely, you have ``git`` and ``shell`` dependencies. I think, from my point of view, its reasonable to say that its on the end-user to modify the code slightly to account for their "atypical" setup.

I hope that does not come off as dismissive. My current aim is to make it so an engine can be added to OpenBench without requiring the contributors donating CPU time to update their clients. Currently, the "bare" minimum requirements are: GCC, G++, Make, an ``EXE=`` hook, a deterministic bench, a way to run the bench (and be parsed by my REGEX), and I suppose the assumption that Makefile path = Output path. I think that is a fair set of requirements.
